snapcraft/plugins/v2/colcon.py

```diff
@@ -0,0 +1,233 @@
+            cmd.extend(["--packages-ignore", *self.options.colcon_packages_ignore])
+
+        if self.options.colcon_packages:
+            cmd.extend(["--packages-select", *self.options.colcon_packages])
+
+        if self.options.colcon_ament_cmake_args:
+            cmd.extend(["--ament-cmake-args", *self.options.colcon_ament_cmake_args])
+
+        if self.options.colcon_catkin_cmake_args:
+            cmd.extend(["--catkin-cmake-args", *self.options.colcon_catkin_cmake_args])
+
+        # Specify the number of workers
+        cmd.extend(["--parallel-workers", "${SNAPCRAFT_PARALLEL_BUILD_COUNT}"])
+
+        return " ".join(cmd)
+
+    def _get_stage_runtime_dependencies_command(self):
+        env = dict(
+            (key, os.environ[key])
+            for key in ["SNAP", "SNAP_ARCH", "SNAP_NAME", "SNAP_VERSION"]
+            if key in os.environ
+        )
+        env["LC_ALL"] = "C.UTF-8"
+        env["LANG"] = "C.UTF-8"
+
+        return " ".join(
+            _get_python_command(
+                env, [os.path.abspath(__file__), "stage-runtime-dependencies"]
+            )
+        )
+
+    def get_build_commands(self) -> List[str]:
+        return [
+            ". /opt/ros/$ROS_DISTRO/setup.sh",
+            "if [ ! -f /etc/ros/rosdep/sources.list.d/20-default.list ]; then sudo rosdep init; fi",
+            "rosdep update --include-eol-distros --rosdistro $ROS_DISTRO",
+            "rosdep install --from-paths . --default-yes --ignore-packages-from-source",
+            self._get_colcon_build_command(),
+            self._get_stage_runtime_dependencies_command(),
+        ]
+
+
+@click.group()
+def plugin_cli():
+    pass
+
+
+@plugin_cli.command()
+@click.option("--part-install", envvar="SNAPCRAFT_PART_INSTALL", required=True)
+@click.option("--ros-distro", envvar="ROS_DISTRO", required=True)
+def stage_runtime_dependencies(part_install: str, ros_distro: str):
+    # XXX: Do we have to stage python packages, or are they all in the repo?
+    apt_packages: Set[str] = set()
+    rosdep_cmd = shutil.which("rosdep")
+    if not rosdep_cmd:
+        rosdep_cmd = "/snap/snapcraft/current/usr/bin/rosdep"
+
+    for pkg in catkin_packages.find_packages(".").values():
+        for dep in pkg.exec_depends:
+            cmd = _get_python_command(
+                dict(), [rosdep_cmd, "resolve", dep.name, "--rosdistro", ros_distro]
+            )
+            try:
+                proc = subprocess.run(
+                    cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+                )
+            except subprocess.CalledProcessError as error:
+                click.echo(f"failed to run {cmd!r}: {error.output}")
+
+            parsed = _parse_rosdep_resolve_dependencies(
+                dep, proc.stdout.decode().strip()
+            )
+            apt_packages |= parsed.pop("apt", set())
+
+            if parsed:
+                click.echo(f"unhandled dependencies: {parsed!r}")
+
+    if apt_packages:
```

Comment:
Do you plan on support pip dependencies at all?

Reply:
I guess we'll have to :)

I don't think it'll be a problem, it just needs some stuff wired up to behave like python plugin that I didn't want to do unless necessary, but it appears to be so.

Reply:
Sorry, didn't intend to start two threads about this. I'll resolve this one and we can continue [here](https://github.com/snapcore/snapcraft/pull/3097#discussion_r442541537).